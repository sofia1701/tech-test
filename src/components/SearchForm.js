import React, { useState } from "react";
import "../styles/SearchForm.css";
import getImages from "../requests/getImages";

const SearchForm = () => {
  const [value, setValue] = useState("");

  const handleChange = (event) => {
    setValue(event.target.value);
  };

  const handleSubmit = (event) => {
    event.preventDefault();
    getImages(value);
  };

  return (
    <div>
      <form className="search-form" onSubmit={handleSubmit}>
        <input
          className="input"
          data-testid="input-id"
          type="text"
          placeholder="Search.."
          value={value}
          onChange={handleChange}
        ></input>
        <button type="submit" className="button">
          Go!
        </button>
      </form>
    </div>
  );
};

export default SearchForm;
