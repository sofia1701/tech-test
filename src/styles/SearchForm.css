.search-form {
  border-radius: 11px;
  font-size: 20px;
  padding: 0 10px;
  text-align: center;
}
.button {
  background-color: red;
  color: antiquewhite;
}
