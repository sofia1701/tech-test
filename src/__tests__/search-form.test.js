import React from "react";
import { render, screen } from "@testing-library/react";
import SearchForm from "../components/SearchForm";

describe("SearchForm", () => {
  it("renders correctly", () => {
    const { asFragment } = render(
      <SearchForm />
    )
    expect(asFragment()).toMatchSnapshot();
  })
  it("renders a text input", () => {
    render(<SearchForm />)
    const input = screen.getByTestId("input-id");

    expect(input).toHaveAttribute("type", "text");
  });
  it("renders a button", () => {
    render(<SearchForm />);
    screen.getByRole('button', {name: /go!/i})
  });
})