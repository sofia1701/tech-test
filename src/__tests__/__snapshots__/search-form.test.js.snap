// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm renders correctly 1`] = `
<DocumentFragment>
  <div>
    <form
      class="search-form"
    >
      <input
        class="input"
        data-testid="input-id"
        placeholder="Search.."
        type="text"
        value=""
      />
      <button
        class="button"
        type="submit"
      >
        Go!
      </button>
    </form>
  </div>
</DocumentFragment>
`;
